File: Database/InitialDBScript/message.js

db.createCollection( 'message', {validator: {$jsonSchema: {bsonType: 'object',title:'message',required: [         'senderId',          'receiverId',          'sentDate',          'text',          'seen'],properties: {senderId: {bsonType: 'objectId'},receiverId: {bsonType: 'objectId'},sentDate: {bsonType: 'timestamp'},text: {bsonType: 'string'},seen: {bsonType: 'bool'}}         }      }});  

File: platformBackend/src/main/java/com/platformBackend/model/entity/Message.java
package com.platformBackend.model.entity;

import lombok.Builder;
import lombok.Data;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@Builder
@Document
public class Message {
    @Id
    private ObjectId _id;
    private ObjectId senderId;
    private ObjectId receiverId;
    @CreatedDate
    private LocalDateTime created;
    @Builder.Default
    private Boolean seen = false;
    private String text;
}
